### DISCUSSION TITLE
Reading LillyMol Reactions into RDKit

### DISCUSSION THREAD

**Discussion by kheyer** [rdkit/rdkit#6142] [Q&A]
Is there a way to read the reaction files from [LillyMol](https://github.com/EliLillyCo/LillyMol) into RDKit? Example reaction files can be found [here](https://github.com/EliLillyCo/LillyMol/tree/master/contrib/data/rxndef), as well as the example reproduced below:

```
(0 Reaction
  (A C Comment "1.2.1_Aldehyde_reductive_amination_FROM_aldehydes_AND_amines")
  (0 Scaffold
     (A C smarts "[OD1]=[CD2T1H]") 
     (A I remove_atom 0)
  )
  (1 Sidechain
     (A C smarts "[ND1H2]-[CX4]||[ND2H]([CX4])[CX4]||[ND1H2]-a||[ND2H]([CX4])-a||[ND2H](a)-a")
     (A I join (1 0))
  )
)
```

SMARTS + atom mapping information is there. Does anyone know a way of getting this file format into the format RDKit uses?

**Comment by greglandrum** [comment]
Hi @kheyer: I'm not aware of any work done to write a parser for that format. It doesn't look like it would be a huge amount of work, but it would depend on how well documented the format is. And getting all the details right would definitely be some work.
